.github/workflows/codecov_ats.yml

```diff
@@ -80,13 +87,43 @@ jobs:
           env:
             CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
             CODECOV_STATIC_TOKEN: ${{ secrets.CODECOV_STATIC_TOKEN }}
-        # Run label analysis IN DRY MODE (no tests will actually run)
-        - name: Label Analysis
+        # Run Automated Test Selection in dry mode to get the list of tests to run
+        # The base commit will be the parent commit (apparently commits on master don't exist in codecov)
+        - name: Codecov Automated Test Selection
+          id: codecov_automated_test_selection
           continue-on-error: true
           run: |
-            BASE_COMMIT=$(git merge-base ${{ github.sha }}^ origin/master)
+            BASE_COMMIT=$(git rev-parse ${{ github.sha }}^)
             echo $BASE_COMMIT
-            codecovcli --codecov-yml-path=codecov.yml label-analysis --dry-run --token=${CODECOV_STATIC_TOKEN} --base-sha=${BASE_COMMIT}
+            output=$(codecovcli --codecov-yml-path=codecov.yml label-analysis --dry-run --token=${CODECOV_STATIC_TOKEN} --base-sha=${BASE_COMMIT}) || true
```

Comment:
To me you are relying on ATS to select the tests to run.
Then in a future step you are relying on pytest to actually run your tests.

Ultimately you are relying on your test results to merge or not merge a PR.
The end goal being "let's merge this PR as fast as possible" or "let's get the go, no-go data for this PR as fast as possible" it's better if ATS can not block your CI.

Yes then it's your problem and responsibility to make sure it's working properly and not just wasting time in your CI, and it might not be ideal in the testing phase for it to fail silently. But overall it's better if it's as transparent as possible.

Ideally we (Codecov) would be able to give you metrics on how ATS is doing for your setup, but we can't do that right now :E 

----

All that being said, if you prefer for this to fail I can make that change.

Reply:
now the whole step can fail